specification/communication/Communication.Messages/models.tsp

```diff
@@ -0,0 +1,322 @@
+
+  @doc("The [Optional] name of the location.")
+  locationName?: string;
+
+  @doc("The [Optional] address of the location.")
+  address?: string;
+
+  @doc("The latitude of the location.")
+  latitude?: float64;
+
+  @doc("The longitude of the location.")
+  longitude?: float64;
+}
+
+@doc("The message template's quick action value information.")
+model MessageTemplateQuickAction extends MessageTemplateValue {
+  @doc("Message parameter type is quick action.")
+  kind: "quickAction";
+
+  @doc("The [Optional] quick action text")
+  text?: string;
+
+  @doc("The [Optional] quick action payload")
+  payload?: string;
+}
+
+@discriminator("kind")
+@doc("The binding object to link values to the template specific locations")
+model MessageTemplateBindings {
+}
+
+@doc("The template bindings for WhatsApp")
+model WhatsAppMessageTemplateBindings extends MessageTemplateBindings {
+  @doc("MessageTemplateBindings is whatsApp")
+  kind: "whatsApp";
+
+  @doc("The header template bindings")
+  header?: WhatsAppMessageTemplateBindingsComponent[];
+
+  @doc("The body template bindings")
+  body?: WhatsAppMessageTemplateBindingsComponent[];
+
+  @doc("The footer template bindings")
+  footer?: WhatsAppMessageTemplateBindingsComponent[];
+
+  @doc("The button template bindings")
+  buttons?: WhatsAppMessageTemplateBindingsButton[];
+}
+
+@doc("The template bindings component for WhatsApp")
+model WhatsAppMessageTemplateBindingsComponent {
+  @doc("The name of the referenced item in the template values.")
+  refValue: string;
+}
+
+@doc("The template bindings component button for WhatsApp")
+model WhatsAppMessageTemplateBindingsButton {
+  @doc("The WhatsApp button sub type")
+  subType?: WhatsAppMessageButtonSubType;
+
+  @doc("The name of the referenced item in the template values.")
+  refValue: string;
+}
+
+@doc("Result of the send message operation.")
+model SendMessageResult {
+  @doc("Receipts of the send message operation.")
+  receipts: MessageReceipt[];
+}
+
+@doc("Receipt of the sending one message.")
+model MessageReceipt {
+  @doc("The message id.")
+  messageId: string;
+
+  @doc("The native external platform user identifier of the recipient.")
+  to: string;
+}
+
+@doc("The WhatsApp-specific template response contract")
+model WhatsAppMessageTemplateItem extends MessageTemplateItem{
+  @doc("""
+WhatsApp platform's template content
+This is the payload returned from WhatsApp
+API.
+""")
+  content?: unknown;
+
+  @doc("Message template response type is whatsApp.")
+  kind: "whatsApp";
+}
+
+@resource("messages/notifications")
+@doc("A notification.")
+model Notifications {
+  @key
+  @visibility("read")
+  @doc("The notification ID.")
+  id: string;
+}
+
+@resource("messages/channels")
+@doc("A communication channel.")
+model CommunicationChannel {
+  @key
+  @visibility("read")
+  @doc("The registration ID of the channel.")
+  channelId: string;
+}
+
+@resource("messages/streams")
+@doc("A data stream.")
+@access(Access.internal)
```

Comment:
Should this be internal for all languages or only C#? Depends on how we're gonna represent this in other languages.